pkgs/applications/misc/input-leap/default.nix

```diff
@@ -0,0 +1,56 @@
+{ lib, fetchFromGitHub, cmake, curl, xorg, avahi, qtbase, qttools, libei ? null, libportal,
+  mkDerivation, pkg-config, openssl, wrapGAppsHook,
+  avahiWithLibdnssdCompat ? avahi.override { withLibdnssdCompat = true; },
+  fetchpatch
+}:
+
+mkDerivation rec {
+  pname = "input-leap";
+  version = "unstable-2023-05-24";
+
+  src = fetchFromGitHub {
+    owner = pname;
+    repo = pname;
+    rev = "5e2f37bf9ec17627ae33558d99f90b7608ace422";
+    hash = "sha256-55RqdRu/Hi2OTiLjAFJ6Gdgg9iO5NIIJCsOkUQjR9hk=";
+    fetchSubmodules = true;
+  };
+
+  buildInputs = [
+    curl qtbase avahiWithLibdnssdCompat
+  ] ++ (with xorg; [
+    libX11 libXext libXtst libXinerama libXrandr libXdmcp libICE libSM
+  ]) ++ lib.optionals (libei != null) [
```

Comment:
```suggestion
  ]) ++ lib.optionals withLibei [
```

Reply:
Where does this `withLibei` variable come from? I don't see it in your suggestions or the original file…

Reply:
The construction `library ? null` and its cousin `if library != null then doThis else doThat` are deprecated.

Instead of it, use `withLibrary` Boolean guard.

Reply:
Sorry, did you mean to include a new `withLibei` parameter in your function parameters suggestion above? Or does it come from somewhere else?
(Should we use the new kebab-case convention here, making it `with-libei`?)

Reply:
Yes, you should include `withLibei` in the inputs.

As far as I remember, Nixpkgs does not use kebab-case in cases like these. Kebab case is more common in software names.